### PR TITLE
Multivariant system test support for TSDB enabled package

### PR DESCRIPTION
**Pull request by agithomas**
- Enhancement

## Proposed commit message

Multivariant system test of TSDB enabled package requires 8.8.1 and above.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

https://github.com/elastic/integrations/issues/8741

## Screenshots

image

### CODE REVIEW

File: docs/developer_tsdb_migration_guidelines.md [hunk 1]
```diff
@@ -153,6 +153,15 @@ A field that holds millions of unique values may not be an ideal candidate for b
 
 # <a id="troubleshooting"></a> Troubleshooting
 
+### Enabling multiple variants support in system tests
+
+If your system test configuration for a TSDB enabled package contains multiple variants, make sure to set the `kibana.version` in the `manifest.yml` file of the package to be `8.8.1` or higher.
```
Comment (ritalwar) [inline docs/developer_tsdb_migration_guidelines.md:158]:
```suggestion
If the system test configuration for a TSDB-enabled package contains multiple variants, make sure to set the `kibana.version` specified in the `manifest.yml` file of the package to `8.8.1` or a version higher.
```

File: docs/developer_tsdb_migration_guidelines.md [hunk 2]
```diff
@@ -153,6 +153,15 @@ A field that holds millions of unique values may not be an ideal candidate for b
 
 # <a id="troubleshooting"></a> Troubleshooting
 
+### Enabling multiple variants support in system tests
```
Comment (ritalwar) [inline docs/developer_tsdb_migration_guidelines.md:156]: Is the Kibana version requirement only for TSDB-enabled packages, or does it apply to any package with multiple variants? If it's specific to TSDB-enabled packages, we can change it to something like: "Configuring system tests for TSDB-enabled packages."

Comment (agithomas) [reply docs/developer_tsdb_migration_guidelines.md:156]: It applies only when you have multi-variants in system tests for a TSDB-enabled package. For other packages, this criteria does not apply.

Comment (ritalwar) [reply docs/developer_tsdb_migration_guidelines.md:156]: I think then we can update this to something like:
```suggestion
### Configuring system tests for TSDB-enabled packages
```

Comment (ritalwar) [reply docs/developer_tsdb_migration_guidelines.md:156]: let's leave it as is since it's TSDB-specific documentation.

**Review by ritalwar — APPROVED** [review]
LGTM!

File: docs/developer_tsdb_migration_guidelines.md [hunk 3]
```diff
@@ -153,6 +153,15 @@ A field that holds millions of unique values may not be an ideal candidate for b
...
+
+If the system test configuration for a TSDB-enabled package contains multiple variants, make sure to set the `kibana.version` specified in the `manifest.yml` file of the package to `8.8.1` or a version higher.
```
Comment (jsoriano) [inline docs/developer_tsdb_migration_guidelines.md:158]: This is actually an issue with any attempt to run the same system tests multiple times in the same deployment. I wonder if we should investigate some way to address this case in `elastic-package`.

I think the issue was fixed in https://github.com/elastic/elasticsearch/pull/96461.

**Review by jsoriano — APPROVED** [review]

**Pull request closed without merging**